### ISSUE TITLE
[Accessibility] Links are missing roles

### ISSUE THREAD

**Issue by natastro** [mozilla-mobile/firefox-ios#23498]
## Prerequisites

- Turn on VoiceOver
- Turn on treatment-a of the New Toolbar iOS experiment
- Found on Firefox Beta iOS app 134.0

## Steps to reproduce

1. Open Firefox Beta iOS app
2. Move VoiceOver focus to the Firefox logo under the URL bar
3. Swipe right through the links in the 4 by 2 grid. Observe VoiceOver output

### Expected behavior / Actual behavior

The expectation is that when VoiceOver lands on a component, it reads to name and the role. The roles in this case would be `link`. At this time, VoiceOver only reads the names

### Device & build information
* Device: iPhone 15, iOS 18.1.1
* Build version: Firefox Beta 134.0
* First seen version: First time I've seen this issue
### Notes
Attachments: 

https://github.com/user-attachments/assets/8dc816b5-6bbe-49fc-b33a-bdcbcf00eed4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-10772)

**Comment by data-sync-user** [comment]
➤ Diana Andreea Barladeanu commented:

Verified as fixed on v135 (60174), with iPhone 16+(18.2).
